src/helpers/MonitorMetrics.js

```diff
@@ -0,0 +1,95 @@
+import { Registry, Gauge } from "prom-client";
+
+export class MonitorMetrics {
+  constructor() {
+    // Create a Registry which registers the metrics
+    this.register = new Registry();
+
+    // Add a default label which is added to all metrics
+    this.register.setDefaultLabels({
+      app: "agoric-cl-oracle-monitor",
+    });
+
+    // Create gauge for value
+    this.oracleSubmission = new Gauge({
+      name: "oracle_latest_value",
+      help: "Latest value submitted by oracle",
+      labelNames: ["oracleName", "oracle", "feed"],
+    });
+
+    // Create gauge for timestamp
+    this.oracleLastEpoch = new Gauge({
+      name: "oracle_last_observation",
+      help: "Last epoch in which oracle made an observation",
+      labelNames: ["oracleName", "oracle", "feed"],
+    });
+
+    // Create gauge for last round
+    this.oracleLastRound = new Gauge({
+      name: "oracle_last_round",
+      help: "Last round in which oracle made an observation",
+      labelNames: ["oracleName", "oracle", "feed"],
+    });
+
+    // Create gauge for price deviation
+    this.oracleDeviation = new Gauge({
+      name: "oracle_price_deviation",
+      help: "Latest price deviation by oracle",
+      labelNames: ["oracleName", "oracle", "feed"],
+    });
+
+    // Create gauge for balance
+    this.oracleBalance = new Gauge({
+      name: "oracle_balance",
+      help: "Oracle balances",
+      labelNames: ["oracleName", "oracle", "brand"],
+    });
+
+    // Create gauge for last price
+    this.actualPriceGauge = new Gauge({
+      name: "actual_price",
+      help: "Actual last price from feed",
+      labelNames: ["feed"],
+    });
+
+    // Register the gauges
+    this.register.registerMetric(this.oracleSubmission);
+    this.register.registerMetric(this.oracleLastEpoch);
+    this.register.registerMetric(this.oracleLastRound);
+    this.register.registerMetric(this.oracleBalance);
+    this.register.registerMetric(this.oracleDeviation);
+    this.register.registerMetric(this.actualPriceGauge);
+  }
+
+  /**
+   * Function to update metrics
+   * @param {*} oracleName oracle name
+   * @param {*} oracle oracle address
+   * @param {*} feed feed name
+   * @param {*} value new feed value
+   * @param {*} id submission id which is a timestamp
+   * @param {*} actualPrice feed actual aggregated price
+   * @param {*} lastRound latest round id for which there was a submission
```

Comment:
These are all of type Gauge

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/672f0f1584e452f9867209af09773164dd2bc949